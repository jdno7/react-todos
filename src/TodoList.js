import todolist from './todos'
import {useState} from 'react'
import NewTodoForm from './NewTodoForm'
// import './TodoList.css'
import {v4 as uuid} from "uuid" 


const TodoList = () => {
    const [todos, setTodos] = useState(todolist)
    const handleRemove = (todo) => {
        const newTodos = todos.filter(td => td != todo)
        setTodos(newTodos)
    }
    const addNewTodo = (newTodo) => {
        setTodos([
            ...todos, newTodo
        ])
    } 

    return (
        <>
        <div>
            <h2>My Todo List</h2>
            {todos.map((td, idx) => 
                <div>{td}
                    <button 
                        onClick={() => handleRemove(td)}
                        key={idx}
                        data-testid={td}>
                        X
                    </button>
                </div>
            )}
            </div>
        <NewTodoForm addNewTodo={addNewTodo}/>
        </>
    )
}

export default TodoList