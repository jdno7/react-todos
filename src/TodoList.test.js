import { getByTestId, render, screen } from '@testing-library/react';
import TodoList from './TodoList';
import {fireEvent} from '@testing-library/react'

test('renders TodoList Component', () => {
  render(<TodoList />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
});

test('Snapshot Test', () => {
    const {asFragment} = render(<TodoList/>)
    expect(asFragment()).toMatchSnapshot()
})

it('should render Todos with remove buttons', () => {
    const {queryAllByText} = render(<TodoList/>)
    expect(queryAllByText("X").length).toEqual(4)
})

it('should remove todo with Click on X Btn', () => {
    const {queryByText, queryByTestId} = render(<TodoList/>)
    expect(queryByText('Do the dishes')).toBeInTheDocument()
    const todoRemoveBtn = queryByTestId('Do the dishes')
    fireEvent.click(todoRemoveBtn)
    expect(queryByText('Do the dishes')).not.toBeInTheDocument()
})

it('Add a new Todo', () => {
    const {queryByTestId, getByLabelText, queryByText} = render(<TodoList/>)
    const todoInput = getByLabelText('Todo')
    const btn = queryByText('Add Todo!')
    
    expect(queryByText('Take out the trash')).not.toBeInTheDocument()
    fireEvent.change(todoInput, {target:{value: "Take out the trash"}})
    fireEvent.click(btn)
    expect(queryByText('Take out the trash')).toBeInTheDocument()
    

})