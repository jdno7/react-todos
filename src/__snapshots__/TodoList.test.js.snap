// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
<DocumentFragment>
  <div>
    <h2>
      My Todo List
    </h2>
    <div>
      Do the dishes
      <button
        data-testid="Do the dishes"
      >
        X
      </button>
    </div>
    <div>
      Laundry
      <button
        data-testid="Laundry"
      >
        X
      </button>
    </div>
    <div>
      Grocery Store
      <button
        data-testid="Grocery Store"
      >
        X
      </button>
    </div>
    <div>
      Do some sex
      <button
        data-testid="Do some sex"
      >
        X
      </button>
    </div>
  </div>
  <div>
    <form>
      <h2>
        Add New Todo
      </h2>
      <label
        for="todo"
      >
        Todo
      </label>
      <input
        id="todo"
        name="todo"
        type="text"
        value=""
      />
      <button>
        Add Todo!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
